src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
+            Actor_MarkForDeath(&this->actor);
+        } else {
+            EnKusa_SetupCut(this);
+            this->isCut = true;
+        }
+
+    } else {
+        if (!(this->collider.base.ocFlags1 & OC1_TYPE_PLAYER) && (this->actor.xzDistToPlayer > 12.0f)) {
+            this->collider.base.ocFlags1 |= OC1_TYPE_PLAYER;
+        }
+
+        if (this->actor.xzDistToPlayer < 600.0f) {
+            Collider_UpdateCylinder(&this->actor, &this->collider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+
+            if (this->actor.xzDistToPlayer < 400.0f) {
+                CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+                if (this->actor.xzDistToPlayer < 100.0f) {
+                    if (GET_KUSA_TYPE(this) != ENKUSA_TYPE_GRASS_2) {
+                        Actor_LiftActor(&this->actor, globalCtx);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void EnKusa_SetupLiftedUp(EnKusa* this) {
+    this->actionFunc = EnKusa_LiftedUp;
+    this->actor.room = -1;
+    this->actor.flags |= 0x10;
+}
+
+void EnKusa_LiftedUp(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f pos;
+    s32 bgId;
+
+    if (Actor_HasNoParent(&this->actor, globalCtx)) {
+        this->actor.room = globalCtx->roomCtx.currRoom.num;
+        EnKusa_SetupFall(this);
+        this->actor.velocity.x = this->actor.speedXZ * Math_SinS(this->actor.world.rot.y);
+        this->actor.velocity.z = this->actor.speedXZ * Math_CosS(this->actor.world.rot.y);
+        this->actor.colChkInfo.mass = 80;
+        this->actor.gravity = -0.1f;
+        EnKusa_UpdateVelY(this);
+        EnKusa_RandScaleVecToZero(&this->actor.velocity, 0.005f);
+        Actor_UpdatePos(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 7.5f, 35.0f, 0.0f, 0xC5);
+        this->actor.gravity = -3.2f;
+    } else {
+        pos.x = this->actor.world.pos.x;
+        pos.y = this->actor.world.pos.y + 20.0f;
+        pos.z = this->actor.world.pos.z;
+        this->actor.floorHeight =
+            BgCheck_EntityRaycastFloor5(&globalCtx->colCtx, &this->actor.floorPoly, &bgId, &this->actor, &pos);
+    }
+}
+
+void EnKusa_SetupFall(EnKusa* this) {
+    this->actionFunc = EnKusa_Fall;
+    rotSpeedXtarget = -0xBB8;
+    rotSpeedYtarget = (Rand_ZeroOne() - 0.5f) * 1600.0f;
+    rotSpeedX = 0;
+    rotSpeedY = 0;
+    this->timer = 0;
+}
+
+/**
+ * @brief Handles logic for when grass/bush falls
+ *
+ */
+void EnKusa_Fall(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 wasHit;
+    Vec3f contactPos;
+    s32 i;
+    s16 angleOffset;
+
+    wasHit = (this->collider.base.atFlags & AT_HIT) != 0;
+    if (wasHit) {
+        this->collider.base.atFlags &= ~AT_HIT;
+    }
+    this->timer++;
+    if (((this->actor.bgCheckFlags & 0xB)) || (wasHit) || ((this->timer) >= 100)) {
+        if ((this->actor.bgCheckFlags & 0x20) == 0) {
```

Comment:
```suggestion
        if (this->actor.bgCheckFlags & 0x20) {
```
Don't need == on flag checks.